src/index.js

```diff
@@ -31,8 +32,14 @@ export const useCourseSearch = (
 ) => {
   const [incremental, setIncremental] = useState(false)
   const [from, setFrom] = useState(0)
-  const [text, setText] = useState("")
-  const [activeFacets, setActiveFacets] = useState(INITIAL_FACET_STATE)
+  const [text, setText] = useState(() => {
+    const { text } = deserializeSearchParams(window.location)
```

Comment:
Hard to tell from the docs but this sets the initial state?

Reply:
yep, if you pass a function to `useState` the return value of that function will be the initial value held in the state container and then on subsequent calls it will just ignore the function.